src/test/java/com/akademia/projectplanner/service/impl/TaskServiceImplTest.java

```diff
@@ -0,0 +1,162 @@
+  private TaskDto taskDto;
+  private AutoCloseable autoCloseable;
+
+  @BeforeEach
+  void setUp() {
+    autoCloseable = MockitoAnnotations.openMocks(this);
+    taskDto = buildTaskDto();
+  }
+
+  @Test
+  void shouldAddTaskSuccessfully() {
+    // given
+    TaskEntity taskEntity = new TaskEntity();
+    Mockito.when(taskMapper.toTaskEntity(taskDto)).thenReturn(taskEntity);
+    // when
+    taskService.addTask(taskDto);
+    // then
+    verify(taskRepository, times(1)).save(taskEntity);
+  }
+
+  @Test
+  void shouldThrowIllegalArgumentExceptionWhenTaskWithBlankName() throws IllegalArgumentException {
+    // given
+    taskDto.setName("");
+    TaskEntity taskEntity = new TaskEntity();
+    Mockito.when(taskMapper.toTaskEntity(taskDto)).thenReturn(taskEntity);
+
+    // when & then
+    assertThrows(IllegalArgumentException.class, () -> taskService.addTask(taskDto));
+  }
+
+  @Test
+  void shouldThrowDateTimeExceptionWhenTaskWithInvalidDate() throws DateTimeException {
+    // given
+    LocalDate today = LocalDate.now();
+    LocalDate yesterday = today.minusDays(1);
+    taskDto.setDeadline(yesterday.toString());
+    TaskEntity taskEntity = new TaskEntity();
+    Mockito.when(taskMapper.toTaskEntity(taskDto)).thenReturn(taskEntity);
+
+    // when & then
+    assertThrows(DateTimeException.class, () -> taskService.addTask(taskDto));
+  }
+
+  @Test
+  void getAllTasks() {
+    // given
+    List<TaskEntity> taskEntities = new ArrayList<>();
+    taskEntities.add(new TaskEntity());
+
+    Mockito.when(taskRepository.findAll()).thenReturn(taskEntities);
+    Mockito.when(taskMapper.toTaskDto(Mockito.any(TaskEntity.class))).thenReturn(taskDto);
+
+    // when
+    List<TaskDto> taskDtos = taskService.getAllTasks();
+
+    // then
+    assertNotNull(taskDtos);
+    assertEquals(taskEntities.size(), taskDtos.size());
+  }
+
+  @Test
+  void shouldGetTaskInfoSuccessfully() {
+    // given
+    TaskEntity taskEntity = new TaskEntity();
+
+    Mockito.when(taskRepository.findById(Mockito.anyLong())).thenReturn(Optional.of(taskEntity));
+    Mockito.when(taskMapper.toTaskDto(Mockito.any(TaskEntity.class))).thenReturn(taskDto);
+
+    // when
+    TaskDto task = taskService.getTaskInfo(1L);
+
+    // then
+    verify(taskRepository, times(1)).findById(1L);
+    assertNotNull(task);
+  }
+
+  @Test
+  void shouldThrowTaskDoesNotExistExceptionWhenTaskNotInRepository()
+      throws TaskDoesNotExistException {
+    // given
+    Long taskId = 1L;
+    Mockito.when(taskMapper.toTaskDto(Mockito.any(TaskEntity.class))).thenReturn(taskDto);
+
+    // when & then
+    assertThrows(TaskDoesNotExistException.class, () -> taskService.getTaskInfo(taskId));
+    verify(taskMapper, never()).toTaskDto(any());
+  }
+
+  @Test
+  void shouldCreateTaskEmailsMapSuccessfully() {
+    // given
+    List<TaskDto> taskList = new ArrayList<>();
+    UserEntity userEntity = new UserEntity();
+    userEntity.setEmail("Email");
```

Comment:
pierdoła, pewnie kwestia sporna, natomiast wyciągnąłbym "Email" do osobnej zmiennej jako String, żeby było tylko jedno miejsce gdzie to się zmienia